canvasobject.go

```diff
@@ -96,3 +96,12 @@ type Focusable interface {
 type Shortcutable interface {
 	TypedShortcut(Shortcut)
 }
+
+// Tabable describes any object that needs to accept the Tab key presses.
+type Tabable interface {
+	// AcceptTabs() is a hook called by the key press handling logic.
+	// If it returns true then the Tab key events will be sent TypedKey and TypedRun.
+	AcceptTabs() bool
+	// SetAcceptTabs() sets the value returned by AcceptTabs().
+	SetAcceptTabs(bool)
```

Comment:
Is this method required? These interfaces are implemented by widgets, should they not be in control of whether tabs are accepted?

Reply:
Do you mean the SetAcceptTabs()? I included it as part of the interface so any widget would have a uniform implementation of turning it on and off. Without it, we would be in a similar state to now, where a widget design decision limits the use of the widget. Having it encourages widget writers to at least consider the tab/no-tab question even if in the end they just implement it as a null function with a suitable comment.
The default state of AcceptTabs() is up to the widget.

Reply:
> Without it, we would be in a similar state to now, where a widget design decision limits the use of the widget.

I don't really follow / agree with this statement. Widget writers are not compelled to use this interface, it is opt-in.
When they decide they want to handle tap then they should implement so that `AcceptsTabs` is covered. exposing `SetAcceptTabs` is in fact leaving the decision open to /app developers/ which is what I don't think we should encourage. If they truly want to override the behaviour they could do so by extension, as I mentioned in the other comment.

Reply:
The tri-state is to allow Entry to maintain its current functionality, but allow it to be overridden and avoid a breaking change. It would be a lot simpler to decouple the tabs from MultiLine and just give Entry a field called say EnableTabs. 
Re SetAcceptTabs, I was looking at it from the contributor's point of view, not an end-user. Not a problem to drop it from the interface now I see your point.

Reply:
> The tri-state is to allow Entry to maintain its current functionality, but allow it to be overridden

Yes I see how it is needed with the current code, but if you remove `SetAcceptTabs` then it is no longer needed becase a developer wishing to change behaviour will override it to make their change.

Reply:
Huh? do I read you right? You want to make it so users have to extend widget.Entry just to turn on/off tab handling?  Isn't that rather tortuous?

Reply:
Our API design is that common use-cases get core API and others are acheived through extending the widget types.

So it hinges on the question - do we want to encourage developers to change how tabs function in standard widgets?

You can compare this to `Cursorable` - it sets a good default and allows developers to change through extension (as it's not super common).

Reply:
Bear in mind that this simple "turn off tabs" is basically changing how the built-in focus handling works and will therefore impact accessibility controls (when it appears) and the general keyboard based control of an app.